casual/casual-caller/src/main/java/se/laz/casual/connection/caller/validation/ConnectionFactoryEntryValidationTimer.java

```diff
@@ -45,13 +54,14 @@ private void setup()
         timerService.createIntervalTimer(0, interval, config);
     }
 
+    @Lock(LockType.READ)
```

Comment:
Does this really work? Reading the docs https://jakarta.ee/specifications/platform/10/apidocs/jakarta/ejb/locktype I get the impression that a READ-lock won't prevent concurrent access unless there is already an active WRITE-lock for the EJB instance `For read-only operations. Allows simultaneous access to methods designated as READ, as long as no WRITE lock is held.`. Or am I missing something? Or does READ type locks still only allow one caller per method and it means something else could call `another` READ-locked method in the same bean as long as no WRITE lock is active?

Reply:
It may indeed not be needed at all in this case as there will ever only be 1 instance of the bean triggering the timeout and in the other bean we handle concurrent calls by ourself.

Reply:
Ok, so how it works is:
Default locking for a Singleton is Write, we do not want that as that could queue up several requests potentially.
However, even with a correct solution, wildfly logs overlapping calls before even calling the timeout method.
Thus I had to go with another solution where we let the work for each time out finish before we schedule another time out.